src/tools/ethBalanceCheck/eth-balance-check.ts

```diff
@@ -53,4 +55,17 @@ export async function getChainInfoByChainID(chainId:string) {
   const allConfigs = await (await fetch("https://chainid.network/chains.json")).json();
   let configInfo = await allConfigs.find((config:any) => config.chainId == parseInt(chainId));
   return configInfo;
+}
+
+export async function getGasWarningAmount(chainName:string, setWarningAmount?:string) { 
+  // Default warning amount
+  let warningAmount = "1";
+  const config = await getConfigByChainName(L1ClientRole.Monitor, chainName);
+  if(config.gasWarningAmount && !isNaN(Number(config.gasWarningAmount))){
+    warningAmount = config.gasWarningAmount;
+  }
+  if(setWarningAmount && !isNaN(Number(setWarningAmount))){
+    warningAmount = setWarningAmount;
+  }
+  return warningAmount
```

Comment:
The order oflogic in this function is not correct. It should be:
let warningAmount = "1";
if(setWarningAmount && !isNaN(Number(setWarningAmount))){
    warningAmount = setWarningAmount;
    return warningAmount ;
  }

 const config = await getConfigByChainName(L1ClientRole.Monitor, chainName);
  if(config.gasWarningAmount && !isNaN(Number(config.gasWarningAmount))){
    warningAmount = config.gasWarningAmount;
  }
return warningAmount

Then this will skip config checking if user had manually set the warning amount, which improve performance and skip unecessary logic.